src/topology/order.lean

```diff
@@ -67,6 +67,36 @@ lemma is_open_generate_from_of_mem {g : set (set α)} {s : set α} (hs : s ∈ g
   @is_open _ (generate_from g) s :=
 generate_open.basic s hs
 
+/-- Consider a set of sets that
+
+- is closed under intersections: `s ∈ g → t ∈ g → s ∩ t ∈ g`;
+- covers the whole space: `(⋃₀ g) = set.univ`.
+
+Then a set is open in the topology generated by this set if and only if it is a union of a set of
+generators. -/
+lemma is_open_generate_from_inter_closed {g : set (set α)} (hg : ∀ s t ∈ g, s ∩ t ∈ g)
+  (hU : (⋃₀ g) = univ) {s} :
+  @is_open _ (generate_from g) s ↔ ∃ S ⊆ g, s = ⋃₀ S :=
```

Comment:
The assumption `hU` is [topological_space.is_topological_basis.eq_generate_from](https://leanprover-community.github.io/mathlib_docs/topology/bases.html#topological_space.is_topological_basis) and `hg` is stronger than `exists_subset_inter`, and the conclusion still holds if you replace `hg` by `exists_subset_inter`: we have [topological_space.is_topological_basis.open_eq_sUnion](https://leanprover-community.github.io/mathlib_docs/topology/bases.html#topological_space.is_topological_basis.open_eq_sUnion). Maybe you could use that or we should do some refactor?

(BTW `hg` is also satisfied by basic opens in the prime spectrum of a commutative ring with the Zariski topology: we have [prime_spectrum.basic_open_mul](https://leanprover-community.github.io/mathlib_docs/algebraic_geometry/prime_spectrum/basic.html#prime_spectrum.basic_open_mul) and [prime_spectrum.is_topological_basis_basic_opens](https://leanprover-community.github.io/mathlib_docs/algebraic_geometry/prime_spectrum/basic.html#prime_spectrum.is_topological_basis_basic_opens).)

Reply:
I guess if you do `letI := generate_from g` then you can prove this lemma using [topological_space.is_topological_basis.open_eq_sUnion](https://leanprover-community.github.io/mathlib_docs/topology/bases.html#topological_space.is_topological_basis.open_eq_sUnion) easily. In your use case (`is_open_iff`) you don't even need `letI`. So I think you can just delete this lemma and use that in `is_open_iff`.

Reply:
I added `is_topological_basis.open_iff_eq_sUnion` and deleted my lemma. Thanks!

Reply:
Thanks! I have no more suggestions or objections, and you already got the delegation, so feel free to merge. But could you add the reference(s) you use, maybe in a future PR, because for example I'm curious to learn more about this topology but couldn't find anything relevant from a brief web search.

Reply:
I didn't use any reference. My goal was to have a topology such that `nhds` is continuous and `tendsto nhds at_top (nhds at_top)`.